dashboard/app/controllers/projects_controller.rb

```diff
@@ -523,6 +525,12 @@ def remix
   def submission_status
     _, project_id = storage_decrypt_channel_id(params[:channel_id])
     project = Project.find_by(id: project_id)
+    begin
+      authorize! :submission_status, project
+    rescue CanCan::AccessDenied => exception
+      error = current_user ? exception.message : SIGN_IN_TO_SUBMIT_PROJECT_MESSAGE
+      return render status: :forbidden, json: {error: error}
```

Comment:
Checking my understanding - now that we check if the user is the project owner in `ability`, couldn't this also happen if the user was signed in but somehow accessed the dialog on a project they didn't own?

Reply:
Yeah, this may be overkill. I checked with Sam about what we should do for signed out users, and she said it was okay to just not show the 'Submit to be featured' button.
On the frontend, planning to not even call on `getSubmissionStatus` if the user is signed out. So I think it may make sense to just remove.
I guess I was thinking of Ben's comment at https://github.com/code-dot-org/code-dot-org/pull/62069#discussion_r1828568314 - but since we're not handling, I think I can remove the customized error msg.

Reply:
Updated!